src/main/java/Duke.java

```diff
@@ -1,10 +1,137 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
 public class Duke {
+    private static Scanner sc;
+    private static List<Task> list;
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        System.out.println("Greetings, I am Nibbalim!\n" + "Your wish is my command!\n");
+        sc = new Scanner(System.in);
+        list = new ArrayList<>();
+        try {
+            Duke.output();
+        } catch (DukeException e) {
+            System.out.println(e);
+        }
+        System.out.println("Kay thanks bye!");
+    }
+
+    public static void output() throws DukeException {
+        while (true) {
+            String input = sc.nextLine();
+            if ("bye".equals(input)) {
+                break;
+            } else if ("list".equals(input)) {
+                System.out.println("Here are the tasks in your list:");
+                for (int i = 0; i < list.size(); i++) {
+                    int j = i + 1;
+                    System.out.println(j + "." + list.get(i).toString());
+                }
+            } else if (isMark(input, list.size())) {
+                int taskindex = Integer.parseInt(input.substring(5)) - 1;
+                list.get(taskindex).markAsDone();
+                System.out.println("Solid! I'm marking this task as done:\n" + list.get(taskindex).toString());
```

Comment:
Line may have been too long, perhaps separate into multiple lines to be more aligned with the coding standard. There are also more similar issues further below.